common/ellipses.js

```diff
@@ -22,39 +22,65 @@
                 };
             }
 
-            if (chaiseConfig.confirmDelete === undefined || chaiseConfig.confirmDelete) {
-                var onError = function (response) {
+            var onError = function (hasConfirm) {
+                return function (response) {
                     scope.$root.showSpinner = false;
                     // if response is string, the modal has been dismissed
-                    if (typeof response !== "string") {
+                    if (!hasConfirm || typeof response !== "string") {
                         ErrorService.handleException(response, true);  // throw exception for dismissible pop- up (error, isDismissible = true)
                     }
                 }
+            }
+
+            var deleteTuple = function (hasConfirmation) {
+                scope.$root.showSpinner = true;
+                // user accepted prompt to delete
+                reference.delete(tuples, logObject).then(function deleteSuccess() {
+                    scope.$root.showSpinner = false;
+                    // tell parent controller data updated
+                    scope.$emit('record-deleted');
+                }).catch(onError());
+            }
+
+            var dataForDelete = {};
+            // if assocation, we have to grab the key information for the association table from
```

Comment:
I know when we say association we always mean pure and binary. But since the implementation of this is actually assuming pure and binary, it's better to emphasis that. Just add a note that you assumed this will be used only for pure and binary association and it will only work for that case (having two foreignkeys).